addons/binding/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/internal/service/TivoStatusProvider.java

```diff
@@ -0,0 +1,723 @@
+                logger.info(" TiVo '{}' - repeating command: '{}' repeating '{}' times",
+                        tivoConfigData.getCfgIdentifier(), pCmd, l);
+            }
+            for (int i = 1; i <= l; i++) {
+                tivoIOSendCommand.println(pCmd.toString() + "\r");
+                if (tivoIOSendCommand.checkError()) {
+
+                    logger.error(" TiVo '{}' - called setTivoStatus and encountered an IO error",
+                            tivoConfigData.getCfgIdentifier(), tivoSocket.isConnected(), tivoSocket.isClosed());
+                    connTivoConnectRetry(false);
+                    connTivoConnectRetry(true);
+
+                    if (retryCount <= 0) {
+                        // We have failed at life... Give up.
+                        return new TivoStatusData(false, -1, "CONNECTION_RETRIES_EXHAUSTED", false, 0);
+
+                    } else {
+                        doNappTime();
+                        return cmdTivoSendRetry(pCmd, retryCount - 1);
+                    }
+                }
+            }
+
+        } catch (Exception e) {
+            logger.error(" TiVo '{}' - message send '{}' failed with error '{}'", tivoConfigData.getCfgIdentifier(),
+                    pCmd.toString(), e.getMessage());
+
+        }
+        // Now read the response
+        return statusGet();
+
+    }
+
+    /**
+     * {@link getTivoStatus} checks if the current connection is "Keep Open" and connected. Connect, query and close as
+     * required.
+     *
+     * @return {@link TivoStatusData} object.
+     */
+
+    private TivoStatusData statusGet() {
+
+        boolean oC = false;
+        TivoStatusData rtnVal = null;
+        int retryCount = 2;
+
+        // Will open a connection prior to operation if we are not keeping one open.
+        if (!connTestSocket() || !tivoConfigData.isCfgKeepConnOpen()) {
+            oC = connTivoConnectRetry(true);
+        }
+
+        if (tivoStatusData.getConnOK() > 1) {
+            rtnVal = statusGetRetry(retryCount);
+        }
+
+        // Will close the connection after operation if we are not keeping one open.
+        if (!tivoConfigData.isCfgKeepConnOpen() && oC && !tivoStatusData.getChScan()) {
+            connTivoConnectRetry(false);
+        }
+        return rtnVal;
+    }
+
+    /**
+     * {@link statusGetRetry} queries the TiVo device for the LAST status message and parses the resultant data.
+     * <p>
+     *
+     * When a command is executed, the system either returns a new CH_STATUS message or any error. Most commands do not
+     * return successful status codes (with the exception of a channel change).
+     * <p>
+     *
+     * @param retryCount integer, number of times a message retrieval will be attempted if a communication error occurs
+     * @return {@link TivoStatusData} object containing the result of the connection to the DVR.
+     */
+    private TivoStatusData statusGetRetry(int retryCount) {
```

Comment:
the same arguments as from `cmdTivoSendRetry` hold true here: the method is called from the `TivoHandler#initialize` method which is meant to return as fast as possible. Here you may again sleep for a significant time.